onnx/shape_inference/implementation.cc

```diff
@@ -629,37 +654,35 @@ void InferShapes(
       schema_registry);
 }
 
-void InferShapes(ModelProto& m, const ISchemaRegistry* schema_registry, const ShapeInferenceOptions& options) {
-  std::unordered_map<std::string, int> opset_imports;
-  for (const auto& opset_import : m.opset_import()) {
-    opset_imports[opset_import.domain()] = static_cast<int>(opset_import.version());
-  }
-
-  ModelLocalFunctionsMap model_local_functions_by_id;
-  for (const auto& function_proto : m.functions()) {
-    model_local_functions_by_id.insert(
-        {GetModelLocalFunctionsMapIdentifier(function_proto.domain(), function_proto.name()), &function_proto});
-  }
-
-  auto* g = m.mutable_graph();
-  SymbolTableImpl symbol_table;
-  TraverseGraphsToAddExistingSymbols(*g, symbol_table);
-  InferShapesImpl(
-      g,
-      std::unordered_map<std::string, TypeProto*>(0),
-      opset_imports,
-      options,
-      &symbol_table,
-      model_local_functions_by_id,
-      schema_registry,
-      m.ir_version());
+void InferShapes(ModelProto& m,
+    const ISchemaRegistry* schema_registry,
+    const ShapeInferenceOptions& options,
+    std::unordered_map<std::string, TensorShapeProto>* generated_shape_data_by_name) {
```

Comment:
What is the specification/contract for this API? I think that to keep this backwards-compatible, the implementation below should handle the case where `(generated_shape_data_by_name == nullptr) && (options.enable_data_propagation )` by creating a local `map` and passing its address to the constructor. I realize that one of the other functions expects the user to do this, but here we are constrained by backwards-compatibility.

Reply:
Okay, never mind, I see that this is happening in `InferShapesImpl` !